File: ProjectTest/OneLang/CSharp_Regen_CSharp/One/Transforms/DetectMethodCalls.cs
using One;
using One.Ast;

namespace One.Transforms
{
    public class DetectMethodCalls : AstTransformer {
        public DetectMethodCalls(): base("DetectMethodCalls")
        {
            
        }
        
        protected override Expression visitExpression(Expression expr) {
            base.visitExpression(expr);
            if (expr is UnresolvedCallExpression && ((UnresolvedCallExpression)expr).func is PropertyAccessExpression)
                return new UnresolvedMethodCallExpression(((PropertyAccessExpression)((UnresolvedCallExpression)expr).func).object_, ((PropertyAccessExpression)((UnresolvedCallExpression)expr).func).propertyName, ((UnresolvedCallExpression)expr).typeArgs, ((UnresolvedCallExpression)expr).args);
            return null;
        }
    }
}

File: ProjectTest/OneLang/Java/src/main/java/OneLang/One/Transforms/DetectMethodCalls/DetectMethodCalls.java
public class DetectMethodCalls extends AstTransformer {
    public DetectMethodCalls()
    {
        super("DetectMethodCalls");
        
    }
    
    protected Expression visitExpression(Expression expr)
    {
        super.visitExpression(expr);
        if (expr instanceof UnresolvedCallExpression && ((UnresolvedCallExpression)expr).func instanceof PropertyAccessExpression)
            return new UnresolvedMethodCallExpression(((PropertyAccessExpression)((UnresolvedCallExpression)expr).func).object, ((PropertyAccessExpression)((UnresolvedCallExpression)expr).func).propertyName, ((UnresolvedCallExpression)expr).typeArgs, ((UnresolvedCallExpression)expr).args);
        return null;
    }
}